### PR TITLE
Feature/se 2071.funding leg notebook

### PR DESCRIPTION
**Pull request by cscialino**
Updates Funding leg notebook.
- Adds a short and long leg.
- Adds Synthetic cash bucket for notional cost
- Adds Coupon payment
- Adds A2B report

### CODE REVIEW

File: examples/use-cases/instruments/FundingLegWithEquity.ipynb [hunk 1]
```diff
@@ -16,7 +16,6 @@
     }
```
Comment (steco) [inline examples/use-cases/instruments/FundingLegWithEquity.ipynb:16]: <p>random newline character</p><p> </p><p>"entering a position *in a swap*"</p><p> </p><p>"transaction type that includes a separate movement for notional cost"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/417/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (steco) [inline examples/use-cases/instruments/FundingLegWithEquity.ipynb:16]: <p>Does this change once ANA-1060 is implemented?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/417/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (steco) [inline examples/use-cases/instruments/FundingLegWithEquity.ipynb:16]: <p>Can we sort out the column headings and order by the SHK values.  The columns should be order start, gains, carry, flows, end</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/417/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (steco) [inline examples/use-cases/instruments/FundingLegWithEquity.ipynb:16]: <pre class="ql-syntax" spellcheck="false">Line #6.        to_effective_at=datetime(2021, 5, 4, tzinfo=pytz.utc),
</pre><p>Can the dates cover 5-6th, ie just one day?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/417/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='48'/>

Comment (cscialino) [reply examples/use-cases/instruments/FundingLegWithEquity.ipynb:16]: <p>Yes we can, though curious why we would do this? Is it to simplify the example?</p><p> </p><p>The longer period includes flows and coupon payments, that range would only show some carry gains.</p>

**Pull request closed without merging**